src/main/java/frc/robot/commands/RotateToNoteAndDriveCommand.java

```diff
@@ -0,0 +1,71 @@
+package frc.robot.commands;
+
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.wpilibj.GenericHID;
+import edu.wpi.first.wpilibj.XboxController;
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.constants.VisionConstants;
+import frc.robot.subsystems.SwerveSubsystem;
+import frc.robot.utils.VisionUtils;
+import java.util.function.DoubleSupplier;
+
+public class RotateToNoteAndDriveCommand extends Command {
+    private SwerveSubsystem swerveSubsystem;
+    private final DoubleSupplier forward;
+    private final DoubleSupplier sideways;
+    private final DoubleSupplier radians;
+    private final ProfiledPIDController rotationPID = new ProfiledPIDController(
+            VisionConstants.rotateToNoteP,
+            VisionConstants.rotateToNoteI,
+            VisionConstants.rotateToNoteD,
+            VisionConstants.rotateToNoteConstraints
+    );
+
+    private final XboxController controller;
+
+    public RotateToNoteAndDriveCommand(SwerveSubsystem swerveSubsystem, XboxController controller, DoubleSupplier forward, DoubleSupplier sideways, DoubleSupplier radians) {
+        this.swerveSubsystem = swerveSubsystem;
+        this.forward = forward;
+        this.sideways = sideways;
+        this.radians = radians;
+        this.controller = controller;
+
+        rotationPID.setTolerance(0.05);
+
+        addRequirements(swerveSubsystem);
+    }
+
+    @Override
+    public void execute() {
+        if (VisionUtils.getNoteTV() == 1.0) {
```

Comment:
I think you want to setup the PID once while not rotating and then let it do it's thing without more feedback from vision. Or alternatively do it in an open loop and just use P.